stock_request_analytic/models/stock_request.py

```diff
@@ -10,28 +10,28 @@ class StockRequest(models.Model):
     _check_company_auto = True
 
     analytic_account_id = fields.Many2one(
-        "account.analytic.account",
+        comodel_name="account.analytic.account",
         string="Analytic Account",
+        compute="_compute_analytic_id",
+        store=True,
+        readonly=False,
         check_company=True,
     )
     analytic_tag_ids = fields.Many2many(
-        "account.analytic.tag",
+        comodel_name="account.analytic.tag",
         string="Analytic Tags",
         check_company=True,
     )
 
-    @api.onchange("product_id")
-    def onchange_product_id(self):
+    @api.depends("order_id")
+    def _compute_analytic_id(self):
         """
         Set default analytic account on lines from order if defined.
         """
-        res = super().onchange_product_id()
-        if self.order_id and self.order_id.default_analytic_account_id:
-            self.analytic_account_id = self.order_id.default_analytic_account_id
-        return res
+        for req in self.sudo():
```

Comment:
@ps-tubtim Have you try to add 'compute_sudo=True' on field directly ?

Reply:
@rousseldenis I already added `compute_sudo=True` on field. Please review it again.